broker-daemon/block-order-worker/index.js

```diff
@@ -258,6 +268,8 @@ class BlockOrderWorker extends EventEmitter {
   async workBlockOrder (blockOrder, targetDepth) {
     this.logger.info('Working block order', { blockOrderId: blockOrder.id })
 
+    if (!blockOrder.isInWorkableState) throw new Error('BlockOrder is not in a state to be worked')
```

Comment:
I don't think we should throw here, for two reasons:

1. This is not an exception per se - if a block order places two orders, one of which gets filled and the other of which ends in an error and fails the block order, the one that gets filled will try to re-work the block order.
2. `workBlockOrder` only ever happens async, so this will just result in an unhandled rejection